examples/widgets/annotated_cursor.py

```diff
@@ -0,0 +1,338 @@
+                             f"be 'x' or 'y'")
+
+        # If position is valid and in valid plot data range.
+        if pos is not None and lim[0] <= pos <= lim[-1]:
+            # Find closest x value in sorted x vector.
+            # This requires the plotted data to be sorted.
+            index = np.searchsorted(data, pos)
+            # Return none, if this index is out of range.
+            if index < 0 or index >= len(data):
+                return None
+            # Return plot point as tuple.
+            return (xdata[index], ydata[index])
+
+        # Return none if there is no good related point for this x position.
+        return None
+
+    def clear(self, event):
+        """
+        Overridden clear callback for cursor, called before drawing the figure.
+        """
+
+        # The base class saves the clean background for blitting.
+        # Text and cursor are invisible,
+        # until the first mouse move event occurs.
+        super().clear(event)
+        if self.ignore(event):
+            return
+        self.text.set_visible(False)
+
+    def _update(self):
+        """
+        Overridden method for either blitting or drawing the widget canvas.
+
+        Passes call to base class if blitting is activated, only.
+        In other cases, one draw_idle call is enough, which is placed
+        explicitly in this class (see *onmove()*).
+        In that case, `~matplotlib.widgets.Cursor` is not supposed to draw
+        something using this method.
+        """
+
+        if self.useblit:
+            super()._update()
+
+
+fig, ax = plt.subplots(figsize=(8, 6))
+
+x = np.linspace(-5, 5, 1000)
+y = x**2
+
+line, = ax.plot(x, y)
+ax.set_xlim(-5, 5)
+ax.set_ylim(0, 25)
+
+# A minimum call
+# Set useblit=True on most backends for enhanced performance
+# and pass the ax parameter to the Cursor base class.
+# cursor = AnnotatedCursor(line=lin[0], ax=ax, useblit=True)
+
+# A more advanced call. Properties for text and lines are passed.
+# Watch the passed color names and the color of cursor line and text, to
+# relate the passed options to graphical elements.
+# The dataaxis parameter is still the default.
+cursor = AnnotatedCursor(
+    line=line,
+    numberformat="{0:.2f}\n{1:.2f}",
+    dataaxis='x', offset=[10, 10],
+    textprops={'color': 'blue', 'fontweight': 'bold'},
+    ax=ax,
+    useblit=True,
+    color='red',
+    linewidth=2)
+
+plt.show()
+
+###############################################################################
+# Trouble with non-biunique functions
+# -----------------------------------
+# A call demonstrating problems with the *dataaxis=y* parameter.
+# The text now looks up the matching x value for the current cursor y position
+# instead of vice versa. Hover your cursor to y=4. There are two x values
+# producing this y value: -2 and 2. The function is only unique,
+# but not biunique. Only one value is shown in the text.
+
+fig, ax = plt.subplots(figsize=(8, 6))
```

Comment:
I would add a (short) title on this Axes, because if you just run the example without looking at it, you might wonder why it's not working. Maybe also on the first figure as well.

Reply:
Done in 68a16913747277b417991cce2e5efd856d36c7a3
I added a ax.set_title("Cursor Tracking [x|y] Position") call for both figures.